### PR TITLE
[SPARK-33347][CORE] Cleanup useless variables of MutableApplicationInfo

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
There are 4 fields in `MutableApplicationInfo ` seems useless:

- `coresGranted`
- `maxCores`
- `coresPerExecutor`
- `memoryPerExecutorMB`

They are always `None` and not reassigned.

So the main change of this pr is  cleanup these useless fields in `MutableApplicationInfo`.

### Why are the changes needed?
Cleanup useless variables.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Pass the Jenkins or GitHub Action

### CODE REVIEW

**Comment by LuciferYang** [comment]
Maybe can delete these 4 variables directly from `ApplicationInfo`, seems they are not `None` only in one test Suite

File: core/src/test/scala/org/apache/spark/deploy/history/ApplicationCacheSuite.scala [hunk 1]
```diff
@@ -128,8 +128,8 @@ class ApplicationCacheSuite extends SparkFunSuite with Logging with MockitoSugar
       completed: Boolean,
       started: Long,
       ended: Long): SparkUI = {
-    val info = new ApplicationInfo(name, name, Some(1), Some(1), Some(1), Some(64),
```
Comment (LuciferYang) [inline core/src/test/scala/org/apache/spark/deploy/history/ApplicationCacheSuite.scala:131]: It seems that only the parameter pass to the constructor in this test case is not `None`

**Comment by dongjoon-hyun** [comment]
cc @HeartSaVioR

**Comment by LuciferYang** [comment]
It seems that these 4 variables in `ApplicationInfo`  are useless also,  Address 294c83c try to clean them up but caused the `MimaExcludes` check failed, so `cff57d8` revert it.

**Comment by HeartSaVioR** [comment]
Looks like `MutableApplicationInfo` was added with all fields in `ApplicationInfo` even it didn't set all of fields, as `ApplicationInfo` was already there. I'm OK to clean up these fields in `MutableApplicationInfo` as it's true we don't set these fields, but you can't change `ApplicationInfo` as it's a part of public API.

**Review by HeartSaVioR — APPROVED** [review]
+1

**Review by maropu — APPROVED** [review]

**Comment by maropu** [comment]
> What does "binary compatibility check" do? Can you explain it for me @dongjoon-hyun ? Thx ~

NOTE: Since the MiMa plugin checks binary compatibility, you need to update a file below  If you change public interfaces;
https://github.com/apache/spark/blob/master/project/MimaExcludes.scala

**Comment by HeartSaVioR** [comment]
retest this, please

**Comment by HeartSaVioR** [comment]
retest this, please

**Comment by LuciferYang** [comment]
> NOTE: Since the MiMa plugin checks binary compatibility, you need to update a file below If you change public interfaces;
> https://github.com/apache/spark/blob/master/project/MimaExcludes.scala

thx @maropu

**Comment by LuciferYang** [comment]
> Looks like MutableApplicationInfo was added with all fields in ApplicationInfo even it didn't set all of fields, as ApplicationInfo was already there. I'm OK to clean up these fields in MutableApplicationInfo as it's true we don't set these fields, but you can't change ApplicationInfo as it's a part of public API.

Thank you for your advice

**Comment by HeartSaVioR** [comment]
retest this, please

**Comment by HeartSaVioR** [comment]
Thanks! Merging to master.

**Comment by HeartSaVioR** [comment]
Thanks for the contribution, I merged into master.

**Comment by LuciferYang** [comment]
thx @maropu @HeartSaVioR @dongjoon-hyun

**Pull request closed without merging**